app/controllers/nonresident/CalculatorController.scala

```diff
@@ -162,7 +162,88 @@ trait CalculatorController extends BaseController {
       else None
     }
 
-    val result = TotalGainModel(flatGain, rebasedGain, timeApportionedGain)
+    TotalGainModel(flatGain, rebasedGain, timeApportionedGain)
+  }
+
+  def calculateTotalGain(disposalValue: Double,
```

Comment:
And whilst not technically your pr that named this method as such but I think something like generateTotalGainsJson may make more sense?